### DISCUSSION TITLE
Programmatically changing a fields value without triggering onChanged callback

### DISCUSSION THREAD

**Discussion by FXschwartz** [danvick/flutter_form_builder#1137] [Q&A]
Is it possible to programmatically change a field's value as you can via - `.currentState?.patchValue([})` and it **not** trigger the fields onChanged callback?

In my fields onChanged callback I set the value to another provider I use. I have to parse the value into different units of measurements.
And when a user changes the units of measurement from inches to feet, for example, I convert them and then update that measurement field using the `.pathValue` method. The problem is that then triggers the onChanged callback, and it ends up messing with my unit's conversions.

Using the standard Flutter TextField and TextEditingController doesn't behave this way. Is there a way for me to do that with this library?

**Comment by deandreamatias** [comment]
HI @FXschwartz 
If shared this code, should be better to understand, but I try answer your question.

Maybe `formKey.currentState?.fields['field_name'].setValue(value)` method can help you, but I don't understand why you want update the form value without update de UI. If form value will be update, the UI should be update to user and then onChanged is trigged

**Reply by FXschwartz** [reply]
Hey @deandreamatias, thanks for the help. I'll create a small-scale example of what my use case is.

But in the meantime, to clarify, I definitely do want to update the UI. I just don't want the onChanged to be triggered if I'm updating the field programmatically.

This is how the built-in Flutter TextField/TextFormField works. If the user is typing in the field, then the onChanged will fire each time, but If I attach a TextEditingController and use it to update the field programmatically, then the onChanged will **not** fire.
